### DISCUSSION TITLE
Prevent rebuilds during capture_screenshots?

### DISCUSSION THREAD

**Discussion by jsorge** [fastlane/fastlane#20342] [Q&A]
The app I'm working on is being localized into many languages and we want to validate how our UI looks using `capture_screenshots`, but running it currently takes about an hour. I see that the project gets rebuilt many times during the process and was hoping I could pair `gym` with it and run snapshots using the `test_without_building` option. The problem I'm running into is that gym seems to really want to make an IPA and not a build that can be useful in the simulator.

Have any of you been able to trim down how much time capturing screenshots takes?

**Comment by timothycosta** [comment]
I wanted to do something similar and came up with something like this to run all languages in a single build:

```
func testAllLanguages() {
  let originalArgs = app.launchArguments
  let languages = ["en", "es", "el", ]
  for lang in languages {
    self.createScreenshots(language: lang, args: originalArgs)
  }
}

func createScreenshots(language: String, args: [String]?) {
    app.launchArguments = args + ["-AppleLanguages", "(\(language))", "-AppleLocale", language]
    app.launch()
    ...
    s?.snapshot("01Library-\(language)") // Screenshot includes language name
}
```

**Reply by jsorge** [reply]
Oh this is interesting! In the snippet above, what is `s?`?. Is that how you're annotating a UI test case?

**Reply by timothycosta** [reply]
It's `Snapshot` defined in `SnapshotHelper.swift` provided by Fastlane.
